ocrd_tesserocr/ocrd-tool.json

```diff
@@ -207,7 +207,7 @@
           "description": "Tesseract OCR engine mode to use:\n* Run Tesseract only - fastest,\n* Run just the LSTM line recognizer. (>=v4.00),\n*Run the LSTM recognizer, but allow fallback to Tesseract when things get difficult. (>=v4.00),\n*Run both and combine results - best accuracy."
         }
       },
-      "resource_locations": ["module"],
+      "resource_locations": ["data"],
```

Comment:
@kba why is that?

Reply:
Because otherwise, the installed models in `$HOME/.local/share/ocrd-resources/ocrd-tesserocr-recognize` were not picked up. What would make sense in general would be `['data', 'module']` so that both the resources in the `data` location as well as module-bundled resources would be found. But that does not make sense for tesserocr either because we need a single place for all the resources.

Reply:
But IIRC the whole point for us was that in Tesseract's case, the resource location is module-defined, because we cannot lookup multiple locations, we must support the TESSDATA_PREFIX envvar, and we also would like to keep supporting the default datadir for Tesseract standalone at build-time.

If `module` is set, then it's actually correct _not_ to look into the `data` path (unless that coincides with the `module` path) IMO. On the contrary: It was an error of `resmgr download` to put the files there in the first place!

In … https://github.com/OCR-D/ocrd_tesserocr/blob/0427f6f31070bdc782a27e71cd6ac88142da9924/ocrd_tesserocr/config.py#L10-L13

… we delegate the `module` directory to `OcrdResourceManager`'s default, which would be the `data` path again. (We could however decide to use the install-time path from Tesseract compile-time here.)

So what we probably missed was to ask the module just where its module path is, so we can **write** new models there. We could also then rewrite the Processor constructor for `--list-resources` and `--show-resource` to **read** models there (so ocrd_tesserocr would not need a custom constructor anymore).

AFAICS all we need is an additional value `module` under `ocrd_utils.RESOURCE_LOCATIONS`, and resolving that value via `Processor.moduledir` and delegating to that in `OcrdResourceManager.location_to_resource_dir`.

Moreover, we should probably change `OcrdResourceManager.default_resource_dir` to query the concrete processor's `resource_locations` and use the first entry of that list instead of just `data`, and also to delegate to that function in `ocrd.cli.resmgr.download` for the `--location` option instead of just `data`.

Reply:
> So what we probably missed was to ask the module just where its module path is,

Yes! That seems to be the missing piece which confused me. Thanks for digging into this, I will open a PR aiming for proper support of `module` location.